### PR TITLE
KSQL-484 - Add version compatibility table to the docs

### PR DESCRIPTION
**Pull request by joel-hamill**
image

- TODO: once approved, add to https://github.com/confluentinc/ksql/pull/504

### CODE REVIEW

**Comment by joel-hamill** [comment]
@miguno @hjafarpour I have rearranged the table to emphasize KSQL version - let me know what you think.

File: docs/README.md [hunk 1]
```diff
@@ -29,4 +29,13 @@ You can use KSQL in standalone, client-server, application, and embedded modes.
 * Beginners: Try the [interactive quick start](/docs/quickstart#quick-start). The quick start configures a single instance in a lightweight Docker container or in a Kafka cluster. It demonstrates a simple workflow using KSQL to write streaming queries against data in Kafka.
 * Advanced users: Try the [end-to-end Clickstream Analysis demo](/ksql-clickstream-demo#clickstream-analysis).
 
+## Interoperability
+
+This table shows the version compatibility matrix of which Kafka clusters can be used to read from and write into while running KSQL queries.
+
+|        KSQL        |        0.1       |
```
Comment (miguno) [inline docs/README.md]: We must also add the latest KSQL release, v0.2.

Comment (joel-hamill) [reply docs/README.md]: @miguno done https://github.com/confluentinc/ksql/pull/526/commits/5013d38d6d56053ee39c148106ebc6f74e22830f

**Comment by miguno** [comment]
@joel-hamill: I just realized that we mention "0.1" and "0.2" in the table, but we also currently use the (for the user) confusing setup where the "0.2" code is actually in the `4.0.x` branch.  Is there a way we could clarify this in the interop table?

**Comment by joel-hamill** [comment]
@miguno we could add another row for GH branch, which corresponds to each version?

**Comment by miguno** [comment]
@joel-hamill : that might be an option.  Perhaps this isn't much a problem any longer once we provide ready-to-use tars/zips for KSQL users that don't want to use docker.  Once these are available, the only people that need to know about branches are contributors to the KSQL code, and we could cover this information elsewhere (rather than making the interop table more confusing for 99% of the users).  WDYT?

**Comment by joel-hamill** [comment]
@miguno yeah i don't think this will be an issue once we move to CP/COS integration. maybe we can wait and see for now?

**Comment by miguno** [comment]
@joel-hamill : Alright, then let's keep it as-is for the time being.

**Pull request merged**